### ISSUE TITLE
Linux 5.18 fails to build

### ISSUE THREAD

**Issue by johnrichardrinehart** [NixOS/nixpkgs#174480]
### Steps To Reproduce
1. Run `nix build github:johnrichardrinehart/johnos#nixosConfigurations.simple-live-iso.pkgs.JohnOS-kernel.kernel`

### Build log
Tail of the build log
```
QUESTION: Memtest, NAME: MEMTEST, ALTS: Y/n/?, ANSWER: y
GOT: y
QUESTION: Microsoft Hyper-V driver testing, NAME: HYPERV_TESTING, ALTS: N/y/?, ANSWER: 
GOT: 
GOT: #
GOT: # configuration written to .config
GOT: #
GOT: make[1]: Leaving directory '/build/linux-5.18/build'
GOT: make: Leaving directory '/build/linux-5.18'
warning: unused option: ASHMEM
error: unused option: DEBUG_INFO
warning: unused option: DEBUG_INFO_BTF
warning: unused option: DEBUG_STACKOVERFLOW
warning: unused option: EXT4_ENCRYPTION
warning: unused option: F2FS_FS_ENCRYPTION
error: unused option: NFSD_V3
warning: unused option: RCU_PERF_TEST
warning: unused option: XEN_SCRUB_PAGES
warning: unused option: XEN_SELFBALLOONING
warning: unused option: XEN_STUB
warning: unused option: XEN_TMEM
```

The build failure seems related to these unused options. I tried removing the corresponding lines (in a local directory - so I changed my `input.nixpkgs` url to the filesystem path). in `/pkgs/os-specific/linux/kernel/common-config.nix` but it didn't fix anything.

### Additional context

It seems that kernel 5.18 phased out some options. Instead of warning us, there's an explicit error. I have no idea how to fix this since `extraConfig` is great for adding lines to the kernel config, but not great at removing them.

### Notify maintainers

@xaverdh @copumpkin @teto (I'm not exactly sure who to notify about this so I just kind of scanned `git blame`s of `common-config.nix` and `top-level/linux-kernels.nix`. Please let me know if there's someone more appropriate to notify.)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Not sure how to do this for this specific package (no `default.nix`).

**Comment by xaverdh** [comment]
Can you check if https://github.com/NixOS/nixpkgs/pull/141505/files fixes this?

**Comment by andrevmatos** [comment]
Should be fixed by #174091
